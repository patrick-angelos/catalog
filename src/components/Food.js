import React from 'react';
import PropTypes from 'prop-types';

const Food = (props) => {
  const {
    id, title, image, cuisine, dish, meal,
  } = props;

  const dishList = dish.map((d) => <li key={`${title}${d}`}>{d}</li>);
  const cuisineText = cuisine || 'world';

  return (
    <div className="food">
      <a className="title" href={`/details/${id}`}>{title}</a>
      <div className="flex space-around"><img className="img" src={image} alt="Sorry, we couldn't find that image" /></div> {/*eslint-disable-line*/}
      <p className="m-t-10">{`Cusine:  ${cuisineText}`}</p>
      <ul className="m-t-10">
        {dishList}
      </ul>
      <p className="m-t-10">{`Best for:  ${meal}`}</p>
    </div>
  );
};

Food.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  cuisine: PropTypes.string,
  dish: PropTypes.arrayOf(PropTypes.string.isRequired),
  meal: PropTypes.arrayOf(PropTypes.string.isRequired),
};

Food.defaultProps = {
  cuisine: 'world',
  dish: [],
  meal: ['anytime'],
};

export default Food;
