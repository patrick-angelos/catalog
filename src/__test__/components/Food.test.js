import React from 'react';
import renderer from 'react-test-renderer';
import Food from '../../components/Food';

describe('Food', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Food
        id="1"
        title="Test meal"
        image="url"
        cuisine="cuisine"
        dish={['dish']}
        meal={['meal']}
        key="1"
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
