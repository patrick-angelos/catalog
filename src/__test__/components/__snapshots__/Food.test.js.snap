// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food renders correctly 1`] = `
<div
  className="food"
>
  <a
    className="title"
    href="/details/1"
  >
    Test meal
  </a>
  <div
    className="flex space-around"
  >
    <img
      alt="Sorry, we couldn't find that image"
      className="img"
      src="url"
    />
  </div>
   
  <p
    className="m-t-10"
  >
    Cusine:  cuisine
  </p>
  <ul
    className="m-t-10"
  >
    <li>
      dish
    </li>
  </ul>
  <p
    className="m-t-10"
  >
    Best for:  meal
  </p>
</div>
`;
